### ISSUE TITLE
PnP Taxonomy Picker slow when loading large termsets and is expanded by default

### ISSUE THREAD

**Issue by krishnanrl** [pnp/sp-dev-fx-controls-react#684]
Thank you for reporting an issue, suggesting an enhancement, or asking a question. We appreciate your feedback - to help the team understand your
needs please complete the below template to ensure we have the details to help. Thanks!

**Please check out the [documentation](https://pnp.github.io/sp-dev-fx-controls-react/) to see if your question is already addressed there. This will help us ensure our documentation is up to date.**

#### Category
[ ] **Question**

#### Version

Please specify what version of the library you are using: [      1.19.0   ]

#### Expected / Desired Behavior / Question
The TermPicker control is very slow while loading a large term set and the terms are all expanded by default. Is there a way of making the terms collapsed by default? I understand that there is an Enterprise Term Picker that only expands terms on demand but that's only to use as a Property Pane control and not as a regular control.
Is there a way to make the Term Picker load faster for large term sets or somehow use the Property control Enterprise Term Picker? 

Thanks!

**Comment by AJIXuMuK** [comment]
Hi @krishnanrl 
Unfortunately, I have to answer no to both of your questions.
I will mark this issue as enhancement to copy the Enterprise taxonomy picker to this library as well.

**Comment by gautamdsheth** [comment]
Hey @AJIXuMuK - was looking at the code at PnP Property controls for porting it over. There we are using PnPJS v1 , whereas here we are using PnPJS v2. 

In PnPJS v2, we don't have full support for taxonomy APIs yet. So, to port this over , should I downgrade it to v1 in this repo along with the enterprise picker control ? I don't think it will break anything for users. 
Once we have full support for Taxonomy APIs in PnPJS v2, we can upgrade both these repos to use it.
Wanted your inputs before I start this :)

**Comment by AJIXuMuK** [comment]
Hi @gautamdsheth,

I'm not a fan of downgrading... I would check if we can use SharePoint `v2` REST API for the metadata instead.

If it's complicated or something is missing - we can downgrade

**Comment by gautamdsheth** [comment]
Totally agree about downgrading. 
The issue is that, if we port over the enterprise picker , it uses the PnPJS v1 to fetch taxonomy data. 
Here, we are using PnPJS v2 which only supports GET operations.
I was planning to re-use the same UI component for this , same approach used in property control. But with PnPJS v2, we wont be able to do so due to this limitation.
Let me do a quick PoC on this, will let you know how it goes.

**Comment by BrianZhangLarsen** [comment]
Agree, this could be a good feature have. Both for the sake of performance, but also for overall UX and overview.
